lib/elixir/lib/map_set.ex

```diff
@@ -84,13 +94,49 @@ defmodule MapSet do
 
   """
   @spec difference(t, t) :: t
+  # If the first set is less than twice the size of the second map,
+  # it is fastest to re-accumulate items in the first set that are not
+  # present in the second set.
```

Comment:
Feel free to say I should remove these comments, I just figured it would be worth while explaining why there are two clauses here.